Source/JavaScriptCore/wasm/WasmParser.h

```diff
@@ -339,6 +339,9 @@ ALWAYS_INLINE bool ParserBase::parseHeapType(const ModuleInformation& info, int3
         return false;
 
     if (heapType < 0) {
+        // anything under 128 would require multiple bytes, which is an invalid type index
+        if (heapType <= -128)
+            return false;
```

Comment:
This solution is fragile.  Let's just make `isValidHeapTypeKind` take an `intptr_t` instead.  It's also weird for `isValidHeapTypeKind` to only take a `TypeKind`, which suggests it's always taking a valid value.  I wonder if there are undefined behavior fallout there.